cmd/flux/main.go

```diff
@@ -152,6 +153,16 @@ func configureKubeconfig() {
 	}
 }
 
+func getDefaultNamespace() string {
+	fromEnv := os.Getenv("FLUX_SYSTEM_NAMESPACE")
+	switch {
+	case fromEnv != "":
+		return fromEnv
+	default:
+		return rootArgs.defaults.Namespace
+	}
+}
```

Comment:
The logged doc default should not take the set env into account I think.

Reply:
I think it should, people should be aware they've changed the default, `--help` would display that.

Reply:
But that no longer makes it display the default, but rather the current configuration. The default is still `flux-system` if the environment variable is `unset`, which is the information I would like to get from documentation.

Now I no longer know what it falls back to if I would `unset`, wile I could still do an `env | grep FLUX_SYSTEM_NAMESPACE` after having read "can be set with [..]".

Reply:
> Now I no longer know what it falls back to if I would unset it

```go
func getDefaultNamespace() string {
	fromEnv := os.Getenv("FLUX_SYSTEM_NAMESPACE")
	switch {
	case fromEnv != "":
		return fromEnv
	default:
		return rootArgs.defaults.Namespace
	}
}
```

it goes to `default` in the switch.

Reply:
But that still requires you to unset the environment variable, which is what I describe above. I am going to need a second opinion on this.

Reply:
This works the same as the `KUBECONFIG` env var for kubectl, if found in env it uses that as the default, if you set the arg to some other value it uses that.

Reply:
Yes, that is how it should work technically which is fine. The point I am not sure about is making the documentation change based on an environment variable that is set, I would expect it to be static.